### PR TITLE
Publish v2 docs to github pages

### PR DESCRIPTION
**Pull request by iam4x**
* `$ yarn dev` -> now runs dev app & docs
* `$ yarn docs:publish` -> build for production & publish docs to gh-pages
* `$ yarn docs:build:dev` -> build locally docs (useful for debugging purpose & used by netlify)

Docs are now built in the `/docs` root folder for any env (no more `/docs` and `/docs-preview`)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ dist/
 dist-es5-module/
 npm-debug.log
 .DS_Store
-docs-preview/
+docs/
+!scripts/docs/
```
Comment (bobylito) [inline .gitignore]: what does the `!` mean?

File: package.json [hunk 2]
```diff
@@ -7,17 +7,15 @@
...
     "build": "./scripts/build.sh",
-    "dev": "./scripts/dev.sh",
+    "dev": "./scripts/dev-docs.sh",
     "staging": "./scripts/staging-deploy.sh",
-    "dev:docs": "./scripts/dev-docs.sh",
     "doctoc": "doctoc --maxlevel 3 README.md CONTRIBUTING.md",
     "lint": "npm run lint:js && npm run lint:css",
     "lint:js": "eslint .",
     "lint:css": "./scripts/lint-css",
-    "docs:jsdoc": "babel-node ./scripts/docs/gen-widget-doc.js",
```
Comment (bobylito) [inline package.json]: We should remove the script itself.

File: .gitignore (hunk 1 above)
Comment (iam4x) [inline .gitignore]: `docs/` match every folder with that name, like for `node_modules/` so we need to exclude from gitignore `/scripts/docs` with `!` 👍

Comment (bobylito) [inline .gitignore]: Ah ok nice 👍

File: package.json (hunk 2 above)
Comment (Haroenv) [inline package.json]: Isn’t this also used in `test-ci.sh`, if so it should be removed there too.

**Pull request merged**